examples/stream-collector/README.md

```diff
@@ -0,0 +1,162 @@
+<!--
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2017 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+## Snap Plugin Library for C++: Collector Plugin Example
+Here you will find an example plugin that covers the basics for writing a collector plugin.
+
+## Plugin Naming, Files, and Directory
+For your collector plugin, create a new repository and name your plugin project using the following format:
+
+>snap-plugin-[plugin-type]-[plugin-name]
+
+For example:
+>snap-plugin-collector-rando
+
+Proposed files and directory structure:  
+```
+snap-plugin-[plugin-type]-[plugin-name]
+ |--src
+  |--[plugin-name].cc  
+  |--[plugin-name]_test.cc  
+```
+
+For example:
+```
+snap-plugin-collector-rando
+ |--src
+  |--rando.cc  
+  |--rando_test.cc  
+```
+
+## Interface methods
+
+In order to write a plugin for Snap, it is necessary to define a few methods to satisfy the appropriate interfaces. These interfaces must be defined for a collector plugin:
+
+```cpp
+/**
+ * The interface for a collector plugin.
```

Comment:
Please update this section with information specific for streaming collector plugin.